include/flatbuffers/flatbuffers.h

```diff
@@ -1235,20 +1282,22 @@ FLATBUFFERS_FINAL_CLASS
   bool force_defaults_;  // Serialize values equal to their defaults anyway.
 
   struct StringOffsetCompare {
-    StringOffsetCompare(const vector_downward &buf) : buf_(&buf) {}
+    StringOffsetCompare(const vector_dir<> &buf) : buf_(&buf) {}
     bool operator() (const Offset<String> &a, const Offset<String> &b) const {
       auto stra = reinterpret_cast<const String *>(buf_->data_at(a.o));
       auto strb = reinterpret_cast<const String *>(buf_->data_at(b.o));
       return strncmp(stra->c_str(), strb->c_str(),
                      std::min(stra->size(), strb->size()) + 1) < 0;
     }
-    const vector_downward *buf_;
+    const vector_dir<> *buf_;
   };
 
   // For use with CreateSharedString. Instantiated on first use only.
   typedef std::set<Offset<String>, StringOffsetCompare> StringOffsetMap;
   StringOffsetMap *string_pool;
 };
+
+typedef FlatBufferBuilderImpl<DOWNWARD> FlatBufferBuilder;
```

Comment:
Please create a test. I don't think what you're doing here can work at all. It would write all elements in reverse order, which would not create a valid FlatBuffer, as a lot of items rely on adjency. See the internals document.